pkg/db/src/types/orders.ts

```diff
@@ -9,10 +9,55 @@ export interface NewOrdersDatabase {
 }
 
 /**
- * A (standardized) customer order interface:
+ * Standardized supplier interface
+ */
+
+export type SupplierInterface<A extends Record<string, any> = {}> = SupplierProto<A> & SupplierData<A>;
+
+/**
+ * Methods interface
+ */
+export interface SupplierProto<A extends Record<string, any> = {}> {
+	create: () => Promise<SupplierInterface<A>>;
+}
+
+/** Data structure */
+export type SupplierData<A extends Record<string, any> = {}> = CouchDocument<
+	{
+		updatedAt: string | null;
+		publishers: string[];
+		displayName: string;
+	} & A
+>;
+/**
+ * Standardized supplier order interface
+ */
+
+export type SupplierOrderInterface<A extends Record<string, any> = {}> = SupplierOrderProto<A> & SupplierOrderData<A>;
+
+/**
+ * Methods interface
+ */
+export interface SupplierOrderProto<A extends Record<string, any> = {}> {
+	create: () => Promise<SupplierOrderInterface<A>>;
+}
+
+/** Data structure */
+export type SupplierOrderData<A extends Record<string, any> = {}> = CouchDocument<
```

Comment:
Let's rename these to supplier batch: `OrderBatchInterface`, `OrderBatchProto`, etc...and drop the confusion between order batch/supplier order (both one in the same)